src/include/duckdb/common/types.hpp

```diff
@@ -434,7 +437,9 @@ struct EnumType{
 
 struct StructType {
 	DUCKDB_API static const child_list_t<LogicalType> &GetChildTypes(const LogicalType &type);
+	DUCKDB_API static child_list_t<LogicalType> &GetChildTypes(LogicalType &type);
 	DUCKDB_API static const LogicalType &GetChildType(const LogicalType &type, idx_t index);
+	DUCKDB_API static LogicalType &GetChildType(LogicalType &type, idx_t index);
```

Comment:
We try to keep types immutable - could we not create a new struct instead of modifying an existing struct in-place?